### PR TITLE
Fix for use after free in logical_fs.cpp #2758

### PR DESCRIPTION
**Pull request by joachimmetz**

### CODE REVIEW

**Comment by joachimmetz** [comment]
@apriestman PTAL

File: tsk/fs/logical_fs.cpp [hunk 1]
```diff
@@ -764,10 +764,11 @@ load_path_from_inum(LOGICALFS_INFO *logical_fs_info, TSK_INUM_T a_addr) {
 	}
 
 	if ((result != TSK_OK) || (!search_helper->target_found)) {
-		free_search_helper(search_helper);
 		tsk_error_reset();
 		tsk_error_set_errno(TSK_ERR_FS_INODE_NUM);
 		tsk_error_set_errstr("load_path_from_inum - failed to find path corresponding to inum %" PRIuINUM, search_helper->target_inum);
```
Comment (simsong) [inline tsk/fs/logical_fs.cpp:769]: That certainly is a use after free.

File: tsk/fs/logical_fs.cpp [hunk 2]
```diff
@@ -764,10 +764,11 @@ load_path_from_inum(LOGICALFS_INFO *logical_fs_info, TSK_INUM_T a_addr) {
...
+                // Free search_helper after using it to format the error string.
+		free_search_helper(search_helper);
```
Comment (simsong) [inline tsk/fs/logical_fs.cpp:771]: and this certainly is the fix.

**Review by simsong — APPROVED** [review]
thanks.

**Pull request merged**